torch/csrc/autograd/variable.cpp

```diff
@@ -22,7 +22,14 @@
 namespace torch {
 namespace autograd {
 Variable::Impl::Impl(at::Tensor data, bool requires_grad, Edge gradient_edge)
-    : TensorImpl(data.type().type_id(), data.type().scalarType(), data.type().allocator(), /* is variable */ true),
+    : TensorImpl(
+        data.type().type_id(),
+        data.type().scalarType(),
+        data.type().allocator(),
+        at::TensorImplOptions(
+          /* is_variable */ true,
```

Comment:
this doesn't look right, shouldn't you use the TensorImplOptions of data except for is_variable?